src/test/java/org/jabref/logic/util/io/FileUtilTest.java

```diff
@@ -477,4 +478,28 @@ void legalPaths(String fileName) {
     void illegalPaths(String fileName) {
         assertTrue(FileUtil.detectBadFileName(fileName));
     }
+
+    @ParameterizedTest
+    @CsvSource({
+            "''                             ,                                     ,   ",
+            "''                             ,                                     , -3",
+            "''                             ,                                     ,  0",
+            "''                             ,                                     ,  3",
+            "''                             ,                                     ,  5",
+            "''                             ,                                     , 10",
+            "''                             , thisisatestfile.pdf                 ,   ",
+            "''                             , thisisatestfile.pdf                 , -5",
+            "''                             , thisisatestfile.pdf                 ,  0",
+            "...                            , thisisatestfile.pdf                 ,  3",
+            "...                            , thisisatestfile.pdf                 ,  5",
+            "th...e.pdf                     , thisisatestfile.pdf                 , 10",
+            "thisisatestfile.pdf            , thisisatestfile.pdf                 , 20",
+            "...                            , longfilename.extremelylongextension ,  5",
+            "...                            , longfilename.extremelylongextension , 10",
+            "...                            , longfilename.extremelylongextension , 20",
```

Comment:
This is not good behavior.

Expected:

```suggestion
            "lo..                          , longfilename.extremelylongextension ,  5",
            "longfil...                   , longfilename.extremelylongextension , 10",
            "longfilename.extr....                            , longfilename.extremelylongextension , 20",
```

Reason: There are plenty !! of characters available at 10 and 20. They should be used. Not just 3